### ISSUE TITLE
After update to GNOME 3.30 unable to launch GNOME DE

### ISSUE THREAD

**Issue by qwelias** [cylgom/ly#56]
After update to GNOME 3.30 unable to launch GNOME DE, but GNOME Classic runs.

When trying to launch GNOME screen flashes with some text for a split second then goes back to login screen

Linux epamrgos 4.18.8-arch1-1-ARCH #1 SMP PREEMPT Sat Sep 15 20:34:48 UTC 2018 x86_64 GNU/Linux

**Comment by cylgom** [comment]
Do you know if you are running Gnome with Wayland or X ? Wayland support is currently experimental, and is no longer available in upcoming version, because I lack the hardware to test it.

**Comment by qwelias** [comment]
@cylgom I'm running X.

I also tried to spot something in `dmesg`, but I'm not particularly good at it, so wasn't able to catch anything useful, feel free to ask for all kinds of logs, if needed.

I have temporarily switched to _*tbsm*_ display manager which launches GNOME option fine, maybe it'll help. Will try to run _*ly*_ a bit later in hopes some package update will fix it.

**Comment by cylgom** [comment]
I will tell you when the new version is released, so you can try again. In the meantime, you can take a look at your ~/.xsession, see if nothing weird lies in it, or read /var/log/Xorg.0.log and see if any (EE) appears somewhere, or send it to me.

**Comment by cylgom** [comment]
Can you try the new version ? It's available :)

**Comment by Boneflame** [comment]
I think I have the similar problem after system (and ly) upgrade. I can't login with **~/.xinitrc**, but with **Xfce4 Session** works, even if in **~/.xinitrc** only include `exec xfce4-session`.



Journal info after login with **~/.xinitrc**:
```
Sep 25 11:33:04 640509-040147 ly[518]: pam_unix(login:session): session opened for user bernie by (uid=0)
Sep 25 11:33:04 640509-040147 systemd[1]: Created slice User Slice of UID 1000.
Sep 25 11:33:04 640509-040147 systemd[1]: Created slice system-user\x2druntime\x2ddir.slice.
Sep 25 11:33:04 640509-040147 systemd[1]: Started /run/user/1000 mount wrapper.
Sep 25 11:33:04 640509-040147 audit[1]: SERVICE_START pid=1 uid=0 auid=4294967295 ses=4294967295 msg='unit=user-runtime-dir@1000 comm="systemd" exe="/usr/lib/systemd/systemd" hostname=? addr>
Sep 25 11:33:04 640509-040147 kernel: audit: type=1130 audit(1537867984.456:41): pid=1 uid=0 auid=4294967295 ses=4294967295 msg='unit=user-runtime-dir@1000 comm="systemd" exe="/usr/lib/syste>
Sep 25 11:33:04 640509-040147 systemd[1]: Starting User Manager for UID 1000...
Sep 25 11:33:04 640509-040147 systemd-logind[502]: New session 1 of user bernie.
Sep 25 11:33:04 640509-040147 systemd[1]: Started Session 1 of user bernie.
Sep 25 11:33:04 640509-040147 systemd[580]: pam_unix(systemd-user:session): session opened for user bernie by (uid=0)
Sep 25 11:33:04 640509-040147 kernel: audit: type=1006 audit(1537867984.466:42): pid=580 uid=0 old-auid=4294967295 auid=1000 tty=(none) old-ses=4294967295 ses=2 res=1
Sep 25 11:33:04 640509-040147 systemd[580]: Listening on Sound System.
Sep 25 11:33:05 640509-040147 kernel: audit: type=1130 audit(1537867984.532:43): pid=1 uid=0 auid=4294967295 ses=4294967295 msg='unit=user@1000 comm="systemd" exe="/usr/lib/systemd/systemd" >
Sep 25 11:33:04 640509-040147 audit[1]: SERVICE_START pid=1 uid=0 auid=4294967295 ses=4294967295 msg='unit=user@1000 comm="systemd" exe="/usr/lib/systemd/systemd" hostname=? addr=? terminal=>
Sep 25 11:33:04 640509-040147 systemd[580]: Reached target Timers.
Sep 25 11:33:04 640509-040147 systemd[580]: Reached target Paths.
Sep 25 11:33:04 640509-040147 systemd[580]: Listening on GnuPG cryptographic agent and passphrase cache (restricted).
Sep 25 11:33:04 640509-040147 systemd[580]: Starting D-Bus User Message Bus Socket.
Sep 25 11:33:04 640509-040147 systemd[580]: Listening on GnuPG network certificate management daemon.
Sep 25 11:33:04 640509-040147 systemd[580]: Listening on GnuPG cryptographic agent and passphrase cache (access for web browsers).
Sep 25 11:33:04 640509-040147 systemd[580]: Listening on p11-kit server.
Sep 25 11:33:04 640509-040147 systemd[580]: Listening on GnuPG cryptographic agent (ssh-agent emulation).
Sep 25 11:33:04 640509-040147 systemd[580]: Listening on GnuPG cryptographic agent and passphrase cache.
Sep 25 11:33:04 640509-040147 systemd[580]: Listening on D-Bus User Message Bus Socket.
Sep 25 11:33:04 640509-040147 systemd[580]: Reached target Sockets.
Sep 25 11:33:04 640509-040147 systemd[580]: Reached target Basic System.
Sep 25 11:33:04 640509-040147 systemd[580]: Reached target Default.
Sep 25 11:33:04 640509-040147 systemd[580]: Startup finished in 57ms.
Sep 25 11:33:04 640509-040147 systemd[1]: Started User Manager for UID 1000.
Sep 25 11:33:06 640509-040147 ly[518]: pam_unix(login:session): session closed for user bernie
```

**Comment by cylgom** [comment]
This is very strange. What distro are you using?

**Comment by qwelias** [comment]
Distro is Antergos

**Comment by cylgom** [comment]
Hi, a few changes were made that could fix that issue. Can you try to compile with the latest commit?

**Comment by qwelias** [comment]
Hi.
built from latest _master_
```
▶ ly --version
ly version 0.2.1.r14.g5e7437b
```
after pressing _enter_ cursor in the top left blinks a few time then **ly** login renders again.

Do you need any logs?

**Comment by LinArcX** [comment]
Same issue. i have arch linux, install the ly just right now. it can login to __gnome-class__, but cannot login to __gnome__.
Gnome version: 3.30.2
Ly version: 0.2.1.r16.g10bd01a

**Comment by genericusername3** [comment]
I'm running Manjaro (technically Arch) and I have the exact same issue. After logging in, the screen flashes black and returns to ly with "logout" in the header. 
```
ly version 0.2.1.r17.gd54e047
Gnome version 3.30.2
```

**Comment by hermann-p** [comment]
I'm running Manjaro, and have the same issue; I cannot launch GNOME wayland (which has been the default for a couple of years), it always returns to the dm login; the "classic" GNOME under X works. This issue also appears when I use lightdm to login; for running GNOME under wayland I have to launch gdm.

**Comment by ZanSara** [comment]
Similar issue here: I have Arch and I can only login to the shell. If I try to login to anything else (I tried Openbox, xfce4 and .xinitrc) ly returns to the login screen with "logout" in the header. LightDM had no issue instead.

**Comment by ZanSara** [comment]
I've dig a little mode into the issue and I found out what was the actual issue for me. The `openbox.desktop` entry (under `/usr/share/xsessions/`) contained this line:
```
Exec=/usr/bin/openbox-session
```
However ly adds another `/usr/bin` to the exec paths, and therefore fails to find the executable. So I installed a clean copy of ly and modified the above line to:
```
Exec=openbox-session
```
After that it was working fine :)

**Comment by cylgom** [comment]
@ZanSara omg thank your for figuring this out. Fixed in 5633128
